cloudferrylib/os/identity/keystone.py

```diff
@@ -617,9 +617,15 @@ def _get_user_tenants_roles_by_db(self, tenant_list, user_list):
                 continue
 
             roles_ids = ast.literal_eval(roles_field)['roles']
+            db_version = self.get_db_version()
+            if db_version in xrange(29, 39):
```

Comment:
Why not simply `29 <= db_version and db_version <= 39`? 2 simple comparisons is much faster than 10 comparisons in a loop. It is also as readable.